it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionControllerISpec.scala

```diff
@@ -342,6 +342,19 @@ class SubscriptionControllerISpec extends BaseControllerISpec with SessionDataMi
         checkHtmlResultWithBodyText(result, htmlEscapedMessage("error.postcode.blacklisted"))
       }
 
+      "initial details are missing" in {
+        AuthStub.hasNoEnrolments(subscribingAgent)
+        sessionStoreService.knownFactsResult = Some(myAgencyKnownFactsResult)
+        sessionStoreService.initialDetails = None
+
+        givenAddressLookupReturnsAddress("addr1")
+        val result = await(controller.submit("addr1")(authenticatedRequest()))
+
+        status(result) shouldBe 303
+        redirectLocation(result).head shouldBe routes.SubscriptionController.showSubscriptionDetails().url
+        sessionStoreService.removeCalled shouldBe true
```

Comment:
If `remove()` has been called then `showSubscriptionDetails` will fail when it tries to `fetchKnownFactsResult` and will just end up calling `sessionMissingRedirect()`. Why not just short circuit this and call `sessionMissingRedirect()` in the first place.

I'm also not sure this test is in the right place (inside `"redisplay form" when`) - all the other tests inside `"redisplay form" when` are redisplaying the form, including data, after a validation error but this isn't what happens in this case.

See `SessionControllerISpec."redirect to the Check Agency Status page if there is no KnownFactsResult in session because the user has returned to a bookmark"` for an example of testing session missing functionality.